arbnode/resource_management.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2023, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
+
+package arbnode
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"regexp"
+	"strconv"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/node"
+	flag "github.com/spf13/pflag"
+)
+
+func InitResourceManagement(conf *ResourceManagementConfig) {
+	if conf.MemoryLimitPercent > 0 {
+		node.WrapHTTPHandler = func(srv http.Handler) (http.Handler, error) {
+			return newResourceManagementHttpServer(srv, newLimitChecker(conf)), nil
+		}
+	}
+}
+
+type ResourceManagementConfig struct {
```

Comment:
If you address the above comment about separate package (e.g. name this as package as resourcemanager) this struct and everything below that contains "ResourceManagement" can just drop that substring since at call sites it will be clear enough e.g. resourcemanager.Config, resourcemanager.newHttpServer, etc...

Short concise names are encouraged in go.

https://google.github.io/styleguide/go/guide#concision
https://google.github.io/styleguide/go/guide#naming

Reply:
Moved to `arbnode/resourcemanager`